### ISSUE TITLE
Allow to add new users to team before accepting invite

### ISSUE THREAD

**Issue by jpmeijers** [grafana/grafana#18292]
**What would you like to be added**:

After adding and inviting a new user I also want to immediately add them to a team. This is so that when they log in for the first time they can immediately see a dashboard with relevant data.

**Why is this needed**:

Saves the time/confusion that arise when a new user accepts an invite but does not see any dashboard. There is currently a back-and-forth between the new user and the admin to create the new user, invite him, user accepts invite, user tells admin, admin sets permissions, admin tells user to log in again.

**Comment by johncave** [comment]
This really breaks users' expectations and creates an admin overhead as well as adding confusion when onboarding customers.

**Comment by Atomicbeast101** [comment]
I would like to follow-up on this request. It seems there has been no activity at all on this. I think this would be a convenient feature to have for both admins and users.

**Comment by jclarijs** [comment]
Bumping this as well.

I have been using Grafana for over a year now, and it is SUCH a pain to add a new user to a team only after they have accepted their invite.

We use teams permissions to show specific dashboards, and if a user has no team, they have no dashboards. It is rather unfriendly to a user to have created an account only to be greeted by an error message saying you have zero permissions.

**Comment by ProteanCode** [comment]
Anything? I am also looking into that

**Comment by zanemountcastle** [comment]
Agreed. We use Grafana to give users access to log data from their hardware deployed in the field. Each group of users has their own team. It's a very unfriendly experience to add a user only for them to have access to nothing until we add them to a team. It would be a really nice feature to include them in a team when inviting a new user.

**Comment by sergey-onanchenko** [comment]
+1 for this.
huge overhead because we need to wait for accepting invite and only then add to team.

**Comment by leanback2020** [comment]
This should be easy to implement. It's way to much overhead to wait for invite to be accepted..

**Comment by Antoine-D6** [comment]
Same for me.
User will validate its account and not see anything.... before the admin will assign its profile to the correct team...
Would like to assign the user to a team before invite acceptation !

**Comment by voiprodrigo** [comment]
Very surprised this can't be done, it's very annoying hassle.

**Comment by batesenergy** [comment]
+1 Please make this a priority. Every user of ours complains about this issue.

**Comment by Codelica** [comment]
Something seems odd here.   All I can think is that the teams feature just must not be very widely used.   When you invite a few dozen people to an organization and then have to randomly add them to their teams as they accept (or in our case as they complain they can't see anything :) ) the pain is real.

**Comment by xandre31** [comment]
Hi grafana dev team,

Is there any chance to see this expected feature to be added soon please ? I agree with all explanations given here and adding a user to a specific team only after its 1st connection is really not user friendly and give a bad feeling for user. After their 1st connection, they are not able to see any dashboards.

Regards,

**Comment by PabblRepos** [comment]
Running into the same problem. I am inviting users but I can't assign them to a team.
Chief @torkelo  ?

**Comment by torkelo** [comment]
@PabblRepos the invite creation does not actually add a user until the invite is used, so tricky to add a user to a team before user is created. We could add a team property (single) to the invite so the user will be added to at least one team when the user signs up using the invite.

**Comment by Atomicbeast101** [comment]
> @PabblRepos the invite creation does not actually add a user until the invite is used, so tricky to add a user to a team before user is created. We could add a team property (single) to the invite so the user will be added to at least one team when the user signs up using the invite. 

How would this work out if there is LDAP authentication in place?

**Comment by shehrozeee** [comment]
> 

can you guide me where to start (what files to look at to understand the flow of invites and what part to modify to add the field to the db and where the teams are in the code.) and I can open a PR when its done.

**Comment by lez-doit** [comment]
Hey, any updates on this? This would be a really useful feature.

**Comment by yabajos1** [comment]
If grafana is installed with ldap or oauth integration, you can pre-create the username and assign it to a team. later when the user with the same name enters for the first time the password is synchronized

**Comment by yabajos1** [comment]
I mean , you can create the user with a random password, and the first login grafana use the ldap/oauth password.
If You need this for many users (create the new user with random pass and assign to team) You can use the grafana api

**Comment by cawhite6** [comment]
+1 for this add, its a logistics nightmare to manage users when inviting them - usually outside clients and vendors in our use case. Saves no time to assign them after getting - "I can't see anything when I log in emails/calls" from account invites. I might as well have created individual accounts to assign to proper TEAM, then tell them to change their passwords. The 2 user account management options are not symbiotic for real user, multi-tenancy management.

100% of the time I know where that user invite needs to be for TEAM folder/permissions. Other aspect of this is a confirmation to admin who sent invite that it was accepted/linked/created would help as well - blah blah accepted your invite, I can go back and immediately add them to the correct TEAM before the calls/emails. Ideally assigning them on invite is preferred but if you can implement a confirmation alert/email quickly as a stop-gap then I'd take that for now.

Been using Grafana for nearly a decade now and this is primary source of pain so far - user management.

**Comment by Maxim11111** [comment]
Hi guys, and who solves the problem? Can I try using the API?

**Comment by Maxim11111** [comment]
Anyway, I did something. These are crutches, of course, but the problem is being solved for me now.  Perhaps it will be useful to someone.
[grafana-team-user-adder](https://github.com/Maxim11111/grafana-team-user-adder)

**Comment by rossenergyllc** [comment]
I'm a new Grafana admin and agree that it would be very, very helpful to specify a team membership when inviting a new user.  Without this, the invite flow is very user and admin unfriendly.  So appreciative of anyone whose capable and willing to work on this.
